shivyc/tree/nodes.py

```diff
@@ -694,42 +696,57 @@ def get_storage(self, spec_kinds, spec_range):
 
         return storage
 
-    def parse_struct_spec(self, node, redec, symbol_table):
-        """Parse a struct ctype from the given decl_nodes.Struct node.
+    def parse_struct_union_spec(self, node, redec, symbol_table):
+        """Parse struct or union ctype from the given decl_nodes.Struct node.
 
-        node (decl_nodes.Struct) - the Struct node to parse
+        node (decl_nodes.Struct/Union) - the Struct or Union node to parse
         redec (bool) - Whether this declaration is alone like so:
 
            struct S;
+           union U;
 
         or declares variables/has storage specifiers:
 
            struct S *p;
            extern struct S;
+           union U *u;
+           extern union U;
 
         If it's the first, then this is always a forward declaration for a
         new `struct S` but if it's the second and a `struct S` already
         exists in higher scope, it's just using the higher scope struct.
         """
         has_members = node.members is not None
+
+        ctype_req = None  # requested CType
```

Comment:
I think you can remove this line, since ctype_req is always set by the next couple lines